deflate.h

```diff
@@ -289,7 +289,7 @@ typedef enum {
  */
 static inline void put_short(deflate_state *s, uint16_t w) {
 #if defined(UNALIGNED_OK)
-    *(uint16_t *)(&s->pending_buf[s->pending]) = w;
+    memcpy(&s->pending_buf[s->pending], &w, 2);
```

Comment:
Instead of checking if `UNALIGNED_OK` is defined, we would need to check if system is little-endian or big-endian.